frontend/src/Plugin.js

```diff
@@ -23,21 +15,49 @@ class ShopgateAnalytics extends SgTrackingPlugin {
    * @param {string} [options.pushToken] The push token of the device.
    * @param {string} [options.access] The access type (App, Web).
    */
-  constructor({ stage, shopNumber, userId, pushToken, access }) {
+  constructor({
+    stage,
+    shopNumber,
+    userId,
+    pushToken,
+    access,
+  }) {
     super('sgAnalytics', { stage });
 
     initSDK(stage);
+    this.config = {};
 
-    sgAnalytics('setConfig', {
+    this.updateConfig({
       shopNumber,
       pushToken,
-      sgUserId: userId,
+      sgUserId: userId ? `${userId}` : userId,
```

Comment:
Does tracking with the `userId` comply with the GDPR? And is it already prepared to delete the tracking data when the user requests it?

Reply:
i contacted our POs